modules/snooker/src/test/ts/browser/TableSizeTest.ts

```diff
@@ -0,0 +1,111 @@
+import * as fc from 'fast-check';
+
+const pixelTableHtml = '<table style="width: 400px"><tbody><tr><td style="width: 200px"></td><td style="width: 200px"></td></tr></tbody></table>';
+const percentTableHtml = '<table style="width: 80%"><tbody><tr><td style="width: 50%"></td><td style="width: 50%"></td></tr></tbody></table>';
+const noneTableHtml = '<table><tbody><tr><td></td><td></td></tr></tbody></table>';
+
+UnitTest.test('TableSize.getTableSize', () => {
+  const pixelTable = Element.fromHtml<HTMLTableElement>(pixelTableHtml);
+  const percentageTable = Element.fromHtml<HTMLTableElement>(percentTableHtml);
+  const noneTable = Element.fromHtml<HTMLTableElement>(noneTableHtml);
+
+  const pixelSizing = TableSize.getTableSize(pixelTable);
+  const percentageSizing = TableSize.getTableSize(percentageTable);
+  const noneSizing = TableSize.getTableSize(noneTable);
+
+  Assert.eq('Pixel sizing detected', 'pixel', pixelSizing.label);
+  Assert.eq('Percentage sizing detected', 'percent', percentageSizing.label);
+  Assert.eq('None sizing detected', 'none', noneSizing.label);
+});
+
+UnitTest.test('TableSize.pixelSizing', () => {
+  const table = Element.fromHtml<HTMLTableElement>(pixelTableHtml);
+  Insert.append(Body.body(), table);
+
+  const sizing = TableSize.getTableSize(table);
+  const warehouse = Warehouse.fromTable(table);
+
+  Assert.eq('Width should be 400px', 400, sizing.width());
+  Assert.eq('Pixel width should be 400px', 400, sizing.pixelWidth());
+  Assert.eq('Cell widths should be 200px each', [ 200, 200 ], sizing.getWidths(warehouse, ResizeDirection.ltr, sizing));
+  Assert.eq('Cell min width should be at least 10px', true, sizing.minCellWidth() >= 10);
+
+  fc.assert(fc.property(fc.integer(-390, 390), fc.integer(400, 1000), (delta, colWidth) => {
+    Assert.eq('Cell delta should be identity', delta, sizing.getCellDelta(delta));
+    Assert.eq('Single column delta width should be the delta', [ delta ], sizing.singleColumnWidth(colWidth, delta));
+  }));
+
+  sizing.setTableWidth(table, [ 100, 100 ], -200);
+  Assert.eq('Table width after resizing is 200px', '200px', Css.getRaw(table, 'width').getOr(''));
+
+  const cell = SelectorFind.descendant<HTMLTableCellElement>(table, 'td').getOrDie();
+  sizing.setElementWidth(cell, 50);
+  Assert.eq('Cell width after resizing is 50px', '50px', Css.getRaw(cell, 'width').getOr(''));
+
+  Remove.remove(table);
+});
+
+UnitTest.test('TableSize.percentageSizing', () => {
+  const container = Element.fromHtml('<div style="position: absolute; left: 0; top: 0; width: 500px"></div>');
+  const table = Element.fromHtml<HTMLTableElement>(percentTableHtml);
+  Insert.append(container, table);
+  Insert.append(Body.body(), container);
+
+  const sizing = TableSize.getTableSize(table);
+  const warehouse = Warehouse.fromTable(table);
+
+  Assert.eq('Width should be 75', 80, sizing.width());
+  Assert.eq('Pixel width should be 600px', 400, sizing.pixelWidth());
```

Comment:
I think the messages are incorrect here.

Reply:
Oops, yep that's some bad refactoring thanks :+1:

Reply:
Fixed